### PR TITLE
Add support for methods with variadic arguments

### PR DESCRIPTION
**Pull request by fvovan**
needed for https://github.com/cargomedia/cm/pull/2540

### CODE REVIEW

**Comment by tomaszdurka** [comment]
I've merged pr about non-dev version of phpcoveralls. Can you please merge upstream?

File: source/CodeGenerator/ParameterBlock.php [hunk 1]
```diff
@@ -58,11 +62,14 @@ public function dump() {
...
         }
+        if ($this->_variadic) {
+            $content .= '...';
+        }
         if ($this->_passedByReference) {
             $content .= '&';
         }
         $content .= '$' . $this->_name;
-        if ($this->_optional) {
+        if ($this->_optional && !$this->_variadic) {
```
Comment (tomaszdurka) [inline source/CodeGenerator/ParameterBlock.php]: Shall we rather throw ? If variadic parameter has default value? Or is optional?

**Comment by fvovan** [comment]
Had to add one more parameter.
The difference between `optional` and `defaultAvailable` is the latter is not supported for builtin  methods. However `optional` parameter can be `variadic` at the same moment from Reflection perspective as it does not necessarily have default value.
@tomaszdurka wdyt?

**Comment by tomaszdurka** [comment]
I thought I replied to this one already.

Do we really need this last commit?
I understand that variadic arguments can be optional, but they cannot have default value. In case of codegenerator we fully ignore `optional` indicator from reflection. The problem could occur before when `optional` was used, but after understanding that it's informational only, I dropped it fully.

So:
Non-variadic, optional params are covered by `isDefaultValueAvailable`.

Did you have a case where it was really needed?

**Comment by fvovan** [comment]
`optional` can't be ignored. some tests require it.
(`CM_Jobdistribution_Job_AbstractTest` and  `CM_JobDistribution_JobWorkerTest` for example)

**Comment by tomaszdurka** [comment]
@fvovan can you check current implementation with Tests you mention?

**Comment by fvovan** [comment]
@tomaszdurka yes, it works.

**Comment by fvovan** [comment]
soooo, shall we merge it?

**Comment by tomaszdurka** [comment]
Sure!

**Pull request merged**